src/analyze-action.ts

```diff
@@ -61,6 +61,13 @@ export async function sendStatusReport(
   await actionsUtil.sendStatusReport(statusReport);
 }
 
+function hasBadExpectErrorInput(): boolean {
+  return (
+    actionsUtil.getOptionalInput("expect-error") === "true" &&
```

Comment:
Perhaps test that this input exists and is set to a truthy value (ie- a non-empty string):

```suggestion
    !!actionsUtil.getOptionalInput("expect-error") &&
```

Reply:
I don't know that this works, because `getOptionalInput` returns `"false"` if the input is unset (because it defaults to false). So if we check if it's truthy, it will return `true` even if it's not been set.

Reply:
I don't know that this works, because `getOptionalInput` returns `"false"` if the input is unset (because it defaults to false). So if we check if it's truthy, it will return `true` even if it's not been set.

Reply:
You can remove the default value perhaps?

Reply:
Or:

```suggestion
    actionsUtil.getOptionalInput("expect-error") !== "false" &&
```
Checks that it is set to a non-default value.

Reply:
Hm, I actually thought that it should be by default `false` so that we wouldn't have to catch the unset/null case in every future use of the variable. Do we prefer the latter?

Reply:
Oops, our comments crossed the wire. I think you're right, and we should just check for a non-default value.

Reply:
Oh nice, I hadn't even thought of that! Thanks. I'll do that.